diff -Nur -x '*.orig' -x '*.rej' e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-ipup mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-ipup
--- e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-ipup	2006-01-11 02:04:56.000000000 -0500
+++ mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-ipup	2007-02-15 12:40:00.000000000 -0500
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 #----------------------------------------------------------------------
-# copyright (C) 2001-2005 Mitel Networks Corporation
+# copyright (C) 2001-2006 Mitel Networks Corporation
 # 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -23,4 +23,4 @@
 #------------------------------------------------------------
 
 /var/qmail/bin/qmail-tcpok
-exec runsvctrl alarm /service/qmail
+exec sv alarm /service/qmail
