diff -Nur -x '*.orig' -x '*.rej' e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/alias/.qmail-shared/50body mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/alias/.qmail-shared/50body
--- e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/alias/.qmail-shared/50body	2006-01-11 18:04:56.000000000 +1100
+++ mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/alias/.qmail-shared/50body	2006-08-14 16:45:40.411590546 +1000
@@ -1,5 +1,13 @@
 {
     use esmith::AccountsDB;
     my $a = esmith::AccountsDB->open_ro;
-    $OUT = join "\n", "&admin", sort map { "&" . $_->key } $a->users;
+
+    $OUT = '';
+
+    for my $user ( $a->get('admin'), $a->users )
+    {
+        next if ( ($user->prop('EveryoneEmail') || 'yes') eq 'no');
+
+        $OUT .= '&' . $user->key . "\n";
+    }
 }
