diff -Nur -x '*.orig' -x '*.rej' e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/control/virtualdomains/80localdomains mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/control/virtualdomains/80localdomains
--- e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/control/virtualdomains/80localdomains	2008-01-08 15:20:14.000000000 -0700
+++ mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/templates/var/qmail/control/virtualdomains/80localdomains	2008-01-08 15:19:27.000000000 -0700
@@ -1,8 +1,6 @@
 {
     $OUT = '';
 
-    my $local_handler = "alias-localdelivery";
-
     use esmith::DomainsDB;
     my $domainsdb = esmith::DomainsDB->open_ro();
 
@@ -14,7 +12,8 @@
 
         next if ( $mail_server ne 'localhost' );
 
-	$domain = $domain->key;
-	$OUT .= "$domain:$local_handler\n";
+	$emaildomain = $domain->key;
+	my $handler = $domain->prop('CatchAll') || "alias-localdelivery";
+	$OUT .= "$emaildomain:$handler\n";
     }
 }
