diff -Nur -x '*.orig' -x '*.rej' e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-update-user mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-update-user
--- e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-update-user	2007-09-06 12:51:46.000000000 -0400
+++ mezzanine_patched_e-smith-qmail-1.10.0/root/etc/e-smith/events/actions/qmail-update-user	2007-09-06 12:51:31.000000000 -0400
@@ -49,9 +49,9 @@
     @users = ( $accountsdb->users, $accountsdb->get('admin') );
 }
 
-foreach my $userName (@users)
+foreach my $user (@users)
 {
-    $userName = $userName->key;
+    my $userName = $user->key;
 
     setpwent();
     my $home = getpwnam($userName)->dir;
